specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-08-03/gallery.json

```diff
@@ -3144,6 +3160,95 @@
           "description": "region of the Gallery Image Version."
         }
       }
+    },
+    "UefiSettings": {
+      "type": "object",
+      "description": "The UEFI settings of the image version.",
+      "properties": {
+        "signatureTemplateNames": {
+          "type": "array",
+          "description": "The name of the template(s) that contains default UEFI key signatures that will be added to the image.",
+          "items": {
+            "$ref": "#/definitions/UefiSignatureTemplateName"
+          },
+          "x-ms-identifiers": []
+        },
+        "additionalSignatures": {
+          "$ref": "#/definitions/uefiKeySignatures"
+        }
+      }
+    },
+    "UefiSignatureTemplateName": {
+      "type": "string",
+      "description": "The name of the signature template that contains default UEFI keys.",
+      "enum": [
+        "NoSignatureTemplate",
+        "MicrosoftUefiCertificateAuthorityTemplate",
+        "MicrosoftWindowsTemplate"
+      ],
+      "x-ms-enum": {
+        "name": "UefiSignatureTemplateName",
+        "modelAsString": true
+      }
+    },
+    "uefiKeySignatures": {
+      "type": "object",
+      "description": "Additional UEFI key signatures that will be added to the image in addition to the signature templates",
+      "properties": {
+        "pk": {
+          "$ref": "#/definitions/uefiKey"
+        },
+        "kek": {
```

Comment:
These property names, kek, pk, db, dbx, are really not helpful or descriptive. I am unhappy to have to release these publicly.

Reply:
Could be better I suppose, but if it makes you feel better, pk, kek, db, dbx aren't really the complete name.  The parent objects provide context for them.  So, don't think about it as pk, think about is as securityProfile.uefiSettings.additionalSignatures.pk, which is already a long name and under that context it's quite clear what pk is if you know about UEFI which, if you're using these properties, you would.
Maybe it also helps that these acronyms are not pulled out of thin air, they're widely used in both MSFT and UEFI public facing docs that talk about TrustedBoot in UEFI.